src/dev/flang/fe/SourceModule.java

```diff
@@ -1655,6 +1655,22 @@ else if (!t1.isDirectlyAssignableFrom(t2, context) &&  // we (currently) do not
     checkDuplicateFeatures(f);
     checkContractAccesses(f);
     checkLegalQualThisType(f);
+    checkLegalDefinesType(f);
+  }
+
+
+  private void checkLegalDefinesType(Feature f)
+  {
+    if (f.definesUsableType() && InTypeFeature(f))
+      {
+        AstErrors.illegalFeatureDefiningType(f);
+      }
+  }
+
+
+  private boolean InTypeFeature(AbstractFeature f)
```

Comment:
method names should start in lower case: `inTypeFeature`:
```suggestion
    if (f.definesUsableType() && inTypeFeature(f))
      {
        AstErrors.illegalFeatureDefiningType(f);
      }
  }


  private boolean inTypeFeature(AbstractFeature f)
```